server/routerlicious/packages/services-client/src/auth.ts

```diff
@@ -7,37 +7,46 @@ import { ITokenClaims, IUser, ScopeType } from "@fluidframework/protocol-definit
 import { KJUR as jsrsasign } from "jsrsasign";
 import jwtDecode from "jwt-decode";
 import { v4 as uuid } from "uuid";
+import { NetworkError } from "./error";
 
 /**
- * Validates a JWT token to authorize routerlicious and returns decoded claims.
- * An undefined return value indicates invalid claims.
+ * Validates a JWT token to authorize routerlicious.
+ * @returns decoded claims.
+ * @throws {NetworkError} if claims are invalid.
  */
 export function validateTokenClaims(
     token: string,
     documentId: string,
-    tenantId: string,
-    maxTokenLifetimeSec: number,
-    isTokenExpiryEnabled: boolean): ITokenClaims {
+    tenantId: string): ITokenClaims {
     const claims = jwtDecode<ITokenClaims>(token);
 
     if (!claims || claims.documentId !== documentId || claims.tenantId !== tenantId) {
-        return undefined;
+        throw new NetworkError(403, "DocumentId and/or TenantId in token claims do not match requested resource");
     }
 
     if (claims.scopes === undefined || claims.scopes.length === 0) {
-        return undefined;
-    }
-
-    if (isTokenExpiryEnabled && claims.exp && claims.iat) {
-        const now = Math.round((new Date()).getTime() / 1000);
-        if (now >= claims.exp || claims.exp - claims.iat > maxTokenLifetimeSec) {
-            return undefined;
-        }
+        throw new NetworkError(403, "Missing scopes in token claims");
     }
 
     return claims;
 }
 
+/**
+ * Validates token claims' iat and exp properties to ensure valid token expiration.
+ * @returns token lifetime in milliseconds.
+ * @throws {NetworkError} if expiry is invalid.
+ */
+export function validateTokenClaimsExpiration(claims: ITokenClaims, maxTokenLifetimeSec: number): number {
```

Comment:
validateTokenExpiration instead of validateTokenClaimsExpiration?

Reply:
Technically it's validating the expiration of TokenClaims being passed to it, and not decoding a token to do so. I feel that validateTokenClaimsExpiration is more accurately descriptive, otherwise I would expect to pass a token in. The reason I don't want to pass the token in, is that we've already decoded it once, so decoding it twice would be an unnecessary potential hit to performance.

Reply:
exp is specifically about token expiring and not about claims expiring. It's just a time to define the timestamp after which the entire token would be invalid. So I'd say its checking token's expiry.

Anyway, I'm not very picky about the name. So either way its fine. :)